torch/csrc/utils/tensor_flatten.h

```diff
@@ -20,8 +20,17 @@ inline std::vector<at::Tensor> unflatten_dense_tensors(const at::Tensor& flat, a
   size_t offset = 0;
   for (const auto & tensor : tensors) {
     auto numel = tensor.numel();
-    outputs.push_back(flat.narrow(0, offset, numel).view(tensor.sizes()));
-    offset += numel;
+    // If unflatten an empty tensor, create a new empty tensor using
+    // flat_tensor_view Options.
+    // This can avoid the unflattened empty tensor to share the same storage
+    // with other unflatten tensors.
+    auto flat_tensor_view = flat.narrow(0, offset, numel).view(tensor.sizes());
+    if (numel == 0) {
+      outputs.push_back(at::empty({0}, flat_tensor_view.options()));
```

Comment:
nit: We don't need flat_tensor_view here right? We can just use flat.options()? Then we only need to create `flat_tensor_view` in the else block below?